### PR TITLE
fix(ci): add cleanup workflow to retain `generated-docs-preview` for only the last 25 PRs

### PR DESCRIPTION
**Pull request by Arqu**
## Change checklist

- [ ] Self-review.
- [ ] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- [ ] Tests if relevant.
- [ ] All breaking changes documented.

### CODE REVIEW

**Review by dignifiedquire — APPROVED** [review]

**Review by matheus23 — CHANGES_REQUESTED** [review]
Ehm, I'm not sure if this will work. The `generated-docs-preview` branch is a "special branch" as far as github is concerned I think.

**Comment by matheus23** [comment]
What exactly is the problem you're trying to solve? Some explanation in the PR description would help me review this!

**Comment by Arqu** [comment]
Thanks for jumping in, I was hoping it works for any branch you set it to (dynamically).
Anyways, the crux of it is this https://github.com/n0-computer/iroh/actions/runs/11105394334/job/30851460274?pr=2756
It seems mostly to be a race between PRs to publish their own. Maybe do a softer version of and do like up to 5 retries with back off.

**Comment by dignifiedquire** [comment]
- what if we switch this to a command instead of auto generating it? eg `/generate_docs`
- alternatively, maybe we just push this onto one of our CI machines and run nginx there just hosting these ourselves per branch

**Comment by matheus23** [comment]
Seems to be a common issue with the action we're using to update the branch: https://github.com/peaceiris/actions-gh-pages/issues/1078

**Comment by matheus23** [comment]
This was recommended as a fix to the concurrency issue: https://github.com/peaceiris/actions-gh-pages/issues/998#issuecomment-1742383037

I don't think our problem is the concurrency though. It seems the action has stopped at some point in the last month or so.

**Comment by matheus23** [comment]
> Thanks for jumping in, I was hoping it works for any branch you set it to (dynamically). Anyways, the crux of it is this https://github.com/n0-computer/iroh/actions/runs/11105394334/job/30851460274?pr=2756 It seems mostly to be a race between PRs to publish their own. Maybe do a softer version of and do like up to 5 retries with back off.

Yeah actually I think this is a separate issue. If pushing to the generated-docs-preview branch were the problem, then the stuff that's *in the branch* would still be served correctly, but e.g. the 2760 PR [exists on the branch](https://github.com/n0-computer/iroh/tree/generated-docs-preview/pr/2760/docs), but [isn't served by github](https://n0-computer.github.io/iroh/pr/2760/docs/iroh/index.html) for some reason.

**Comment by Arqu** [comment]
Ok, think I've figured it out. Seems like we're hitting some file size limits judging by the action outputs. I'll manually prune some of the old PRs and see if that makes things work again.

**Comment by Arqu** [comment]
Yup that worked just fine now. I'll add a step to clean up the branch every now and then.

**Comment by Arqu** [comment]
@matheus23 This should now also trigger a weekly job to clean up the docs branch. It does risk to remove the preview for PRs that are old but still open, but I guess if you still need or want the docs preview you can easily re-run the job to generate them.

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -3,6 +3,10 @@ name: Docs Preview
 on:
   pull_request:
```
Comment (matheus23) [inline .github/workflows/docs.yaml:4]: Can you add `workflow_dispatch` here, too? So we can trigger it manually?

**Review by matheus23 — APPROVED** [review]
Thank you for figuring it out! :)
Just a small nit.

**Comment by Arqu** [comment]
Think this should do it, but cant be 100% sure till we merge and test it out due to the nature of workflows and permissions on GHA.

**Comment by Arqu** [comment]
> Why does netsim keep adding comments? 😅 (Is that intentional?)

It's supposed to update them but something is messed up, that's what I'm going to fix next.

**Pull request merged**